packages/backend-common/src/database/sqlite3.ts

```diff
@@ -29,6 +31,20 @@ export function createSqliteDatabaseClient(
   overrides?: Knex.Config,
 ) {
   const knexConfig = buildSqliteDatabaseConfig(dbConfig, overrides);
+
+  // If storage on disk is used, ensure that the directory exists
+  if (
+    typeof knexConfig.connection === 'object' &&
+    (knexConfig.connection as Knex.Sqlite3ConnectionConfig).filename
```

Comment:
hm so we check here again - this duality of object vs string is odd, maybe this could be made better somehow 🤔  maybe just a util function `getConfigFilename` that returns a string or undefined?

Reply:
This would be less of an issue if I normalize the override too, then I only need to check for null here. Or I even always initialize it to an empty object.

I updated the PR with these two changes.